Sources/Rendering/OpenGL/RenderWindow/index.js

```diff
@@ -90,6 +90,59 @@ function vtkOpenGLRenderWindow(publicAPI, model) {
   // Set our className
   model.classHierarchy.push('vtkOpenGLRenderWindow');
 
+  function createContextHandler() {
+    const cache = new Map();
+
+    const getParameterHandler = {
+      apply(target, gl, args) {
+        if (cache.has(args[0])) {
+          return cache.get(args[0]);
+        }
+        return target.apply(gl, args);
+      },
+    };
+
+    // only supports single-value setters
+    function cachedSetterHandler(key) {
+      return {
+        apply(target, gl, args) {
+          cache.set(key, args[0]);
+          return target.apply(gl, args);
+        },
+      };
+    }
+
+    const propHandlers = {
+      getParameter(gl, prop, receiver, propValue) {
+        return new Proxy(propValue.bind(gl), getParameterHandler);
+      },
+      depthMask(gl, prop, receiver, propValue) {
+        return new Proxy(
+          propValue.bind(gl),
+          cachedSetterHandler(gl.DEPTH_WRITEMASK)
+        );
+      },
+    };
+
+    return {
+      get(gl, prop, receiver) {
+        let ret = Reflect.get(gl, prop, receiver);
+        if (ret instanceof Function) {
+          // prevents Illegal Invocation errors
+          ret = ret.bind(gl);
+        }
+
+        if (prop in propHandlers) {
+          return propHandlers[prop](gl, prop, receiver, ret);
```

Comment:
Performance being important here, instead of checking if `prop` belongs to `propHandlers` and then get the prop handler, how about:
```
const propHandler = propHandles[prop];
if (propHandler) {
  return propHandler(...)
}
```